Formula/mdoc.rb

```diff
@@ -0,0 +1,41 @@
+class Mdoc < Formula
+  desc "Command-line tool for writing scientific documents"
+  homepage "https://kmaasrud.com/mdoc"
+  url "https://github.com/kmaasrud/mdoc/archive/0.3.1.tar.gz"
+  sha256 "6807701201e0149bdb09d2ca7f32590a646dfb238823a5f448384a844e8b78f1"
+  license "MIT"
+  head "https://github.com/kmaasrud/mdoc.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "freetype"
+  depends_on "graphite2"
+  depends_on "harfbuzz"
+  depends_on "libpng"
+  depends_on "openssl@1.1"
+  depends_on "pandoc"
+  uses_from_macos "icu4c"
+
+  def install
+    # From https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/tectonic.rb
+    ENV.cxx11
+    ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version # needed for CLT-only builds
+    ENV.delete("HOMEBREW_SDKROOT") if MacOS.version == :high_sierra
```

Comment:
Why?

Reply:
MDoc depends on Tectonic, which needs quite a few external C libraries included. This is to ensure they are properly linked to when building. As mentioned in the comment, I found this method from the Tectonic formula, and it worked fine on my computer 😄

Reply:
Which external C libraries are these?

Reply:
The dependencies are documented [here](https://tectonic-typesetting.github.io/book/latest/howto/build-tectonic/index.html#third-party-dependencies). You can see the Tectonic guys doing the same on [their formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tectonic.rb).

Reply:
What happens when you don't do this? I can see the `tectonic` formula does the same, but I'm not even sure `tectonic` should be doing that at all.